internal/bson2/raw_document.go

```diff
@@ -40,6 +40,28 @@ func (doc *RawDocument) LogValue() slog.Value {
 // nested documents and arrays are converted to RawDocument and RawArray respectively,
 // using raw's subslices without copying.
 func (raw RawDocument) Decode() (*Document, error) {
+	res, err := raw.decode(false)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	return res, nil
+}
+
+// DecodeDeep decodes a single BSON document that takes the whole raw slice.
+//
+// All nested documents and arrays are decoded recursively.
+func (raw RawDocument) DecodeDeep() (*Document, error) {
```

Comment:
Probably not, because it will make the usage much harder: what happens when some field is already present in the argument? will it be replaced? merged somehow? what happens if the field is present in the argument, but not in the receiver? etc